service/models/operations/optimize.py

```diff
@@ -151,6 +152,11 @@ def gen_pyciemss_args(self, job_id):
         if step_size is not None and solver_method == "euler":
             solver_options["step_size"] = step_size
 
+        total_possible_iterations = (
+            extra_options.get("maxiter") + 1
+        ) * extra_options.get("maxfeval")
+        progress_hook = OptimizeHook(job_id, total_possible_iterations)
```

Comment:
Are we guaranteed to have maxiter and maxfeval always defined when we reach this stage?

Reply:
If we do not provide them via our client's call (which we will) they have defaults in `pyciemss-service` 
I cant think of how they would not be defined but maybe i am missing something so I can surround this by a try catch

Reply:
Okay, should be fine as-is, just checking